spinta/types/type.py

```diff
@@ -43,7 +44,10 @@ def is_valid(self, value):
 class Date(Type):
 
     def load(self, value: typing.Any):
-        return date.fromisoformat(value)
+        try:
+            return date.fromisoformat(value)
+        except ValueError:
+            raise DataError(f"Expected date value in ISO 8601 format ('YYYY-MM-DD'), got {value!r}")
```

Comment:
By the way, what error message is given by Python? Maybe it would be even better to use error message generated by Python standard library?

```python
except ValueError as e:
    raise DataError(f"Invalid date value: {e}.")
```

Reply:
```
In [1]: import datetime

In [2]: datetime.datetime.fromisoformat('2019-04')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-03964cd7b0d8> in <module>
----> 1 datetime.datetime.fromisoformat('2019-04')

ValueError: Invalid isoformat string: '2019-04'
```

It's concise and not as explicit, but maybe good enough?

Reply:
I think it is good enough. Maybe there are other cases, where different error messages are given. So probably using original error messages will be more informative.